### PR TITLE
findbugs: fix default encoding issues

### PR DESCRIPTION
**Pull request by mwinter69**
Get the charset of a run and use it whenever we need to convert the String to a byte array.

### CODE REVIEW

File: src/main/java/jenkins/plugins/logstash/persistence/AbstractLogstashIndexerDao.java [hunk 1]
```diff
@@ -56,6 +58,17 @@ public AbstractLogstashIndexerDao(String host, int port, String key, String user
     }
   }
 
+  @Override
+  public void setCharset(Charset charset)
```
Comment (jakub-bochenski) [inline src/main/java/jenkins/plugins/logstash/persistence/AbstractLogstashIndexerDao.java]: Can you make it explicit what the purpose here is? We have the charset using during the build, and the charset used to encode the outgoing messages. 

Those two do not necessarily need to be the same.

**Comment by jakub-bochenski** [comment]
For backends that use HTTP transport (Logstash, ES) it would be nice to use the `charset` mime type parameter (e.g. `application/json; charset=UTF-8` for the output messages.
Do you think this is feasible?

File: src/main/java/jenkins/plugins/logstash/persistence/AbstractLogstashIndexerDao.java (hunk 1 above)
Comment (mwinter69) [reply src/main/java/jenkins/plugins/logstash/persistence/AbstractLogstashIndexerDao.java]: Added javadoc to describe the purpose

**Comment by mwinter69** [comment]
For this PR I would keep as is. Making mimetype and charset configurable should be done once we have completed the refactoring of configuration

**Pull request merged**